src/commands/aksCreateCluster/aksCreateCluster.ts

```diff
@@ -0,0 +1,176 @@
+import SubscriptionTreeItem from '../../tree/subscriptionTreeItem';
+import { ResourceIdentityType } from '@azure/arm-containerservice';
+import { longRunning } from '../utils/host';
+import { ResourceGroupsListNextResponse } from '@azure/arm-resources/esm/models';
+const meta = require('../../../package.json');
+
+
+interface State {
+    title: string;
+    step: number;
+    totalSteps: number;
+    resourceGroup: QuickPickItem;
+    name: string;
+    clustername: string;
+    subid: string | undefined;
+}
+
+/**
+ * A multi-step input using window.createQuickPick() and window.createInputBox().
+ * 
+ * This first part uses the helper class `MultiStepInput` that wraps the API for the multi-step case.
+ */
+export default async function aksCreateCluster(
+    context: IActionContext,
+    target: any
+): Promise<void> {
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+
+    const cluster = getAksClusterSubscriptionItem(target, cloudExplorer);
+    if (failed(cluster)) {
+        vscode.window.showErrorMessage(cluster.error);
+        return;
+    }
+
+    async function collectInputs() {
+        const state = {} as Partial<State>;
+        await MultiStepInput.run(input => inputClusterName(input, state));
+        return state as State;
+    }
+
+    const title = 'Create AKS Cluster';
+
+    async function inputClusterName(input: MultiStepInput, state: Partial<State>) {
+        state.clustername = await input.showInputBox({
+            title,
+            step: 1,
+            totalSteps: 2,
+            value: state.clustername || '',
+            prompt: 'Choose a unique name for the AKS cluster \n (Valid cluster name is 1 to 63 in length consist of letters, numbers, dash and underscore)',
+            validate: validateAKSClusterName,
+            shouldResume: shouldResume
+        });
+
+        return (input: MultiStepInput) => pickResourceGroup(input, state);
+    }
+
+    const resourceList = await getResourceGroupList(<SubscriptionTreeItem>cluster.result);
+    if (!resourceList.succeeded) {
+        return
+    }
+    const resourceNameGroupList = getResourceNameGroupList(resourceList.result);
+    const resourceGroups: QuickPickItem[] = resourceNameGroupList.map(label => ({ label }));
+
+    async function pickResourceGroup(input: MultiStepInput, state: Partial<State>) {
+        const pick = await input.showQuickPick({
+            title,
+            step: 2,
+            totalSteps: 2,
+            placeholder: 'Pick a resource group',
+            items: resourceGroups,
+            enabled: false,
+            activeItem: typeof state.resourceGroup !== 'string' ? state.resourceGroup : undefined,
+            shouldResume: shouldResume
+        });
+
+        state.resourceGroup = pick;
+    }
+
+
+    function shouldResume() {
+        // Could show a notification with the option to resume.
+        return new Promise<boolean>((resolve, reject) => {
+            // noop
```

Comment:
I think this will never complete. This gets called when an input (text/dropdown) is cancelled (`onDidHide`). So what actually happens is the main Promise in `showInputBox` and `showQuickPick` never completes, and the `finally` block never gets hit, and the disposables are never cleaned up.

We _could_ change it to `return Promise.resolve(false);`... But then you run into another problem. `MultiStepInput.run()` completes without an error, even when the user escapes out, and we go on to try and create the cluster, even though the user didn't want to.

I can see this code came directly from the Microsoft [samples](https://github.com/Microsoft/vscode-extension-samples/blob/main/quickinput-sample/src/multiStepInput.ts), but I think that's where the real problem is. Let me make some tweaks...

Reply:
Perfect!! Pete and I had conversation already and this is something we will rearrange code for, Thanks heaps Pete.